ndc-reference/bin/reference/main.rs

```diff
@@ -1541,19 +1589,100 @@ fn eval_dimension(
             arguments,
             field_path,
             path,
-        } => eval_column_at_path(
-            collection_relationships,
-            variables,
-            state,
-            row,
-            path,
-            column_name,
-            arguments,
-            field_path.as_deref(),
-        ),
+            extraction,
+        } => {
+            let value = eval_column_at_path(
+                collection_relationships,
+                variables,
+                state,
+                row,
+                path,
+                column_name,
+                arguments,
+                field_path.as_deref(),
+            )?;
+
+            eval_extraction(extraction, value)
+        }
     }
 }
 // ANCHOR_END: eval_dimension
+// ANCHOR: eval_extraction
+fn eval_extraction(
+    extraction: &Option<ndc_models::ExtractionFunctionName>,
+    value: serde_json::Value,
+) -> Result<serde_json::Value> {
+    match extraction {
+        None => Ok(value),
+        Some(extraction) => {
+            let iso8601::Date::YMD { year, month, day } = iso8601::date(value.as_str().ok_or({
```

Comment:
```suggestion
            let iso8601::Date::YMD { year, month, day } = iso8601::date(value.as_str().ok_or_else(|| {
```

Use a closure to avoid creating the error data unless there's actually an error. No lazy evaluation in Rust! 😢